embed-code-go/embedding/processor.go

```diff
@@ -207,6 +139,59 @@ func CheckUpToDate(config configuration.Configuration) {
 	}
 }
 
+// Creates and returns new Context based on Processor.DocFilePath and Processor.Config.
+//
+// If any problems faced, an error is returned.
+//
+// Processes an embedding by iterating through different states based on transitions until it
+// reaches the finish state. If a transition is recognized, it updates the current state and
+// accepts the transition. If no transition is accepted, the error indicating the failure to parse
+// the document file is returned.
+func (p Processor) constructEmbedding() (parsing.Context, error) {
```

Comment:
Can we find a better naming. Now it `constructEmbedding ` and I expect `Embeding` or something similar to be returned while here `Context` is returned.